classes/class-p4-master-site.php

```diff
@@ -605,6 +606,30 @@ public function view_meta_box_search( $post ) {
 		<?php
 	}
 
+	/**
+	 * Forces a user to login using Google Auth if they have a greenpeace.org email
+	 *
+	 * @param WP_User $user The current user logging in.
+	 * @param String  $username The username of the user.
+	 * @param String  $password The password of the user.
+	 */
+	public function enforce_google_signon( $user, $username, $password ) {
+
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG === true ) {
+			return $user;
+		}
+
+		if ( is_wp_error( $user ) ) {
+			return $user;
+		}
+
+		if ( strpos( $user->data->user_email, '@greenpeace.org' ) !== false ) {
```

Comment:
`=== false` :)

If you want to be exact (in case somebody ever uses a `greenpeace.org` subdomain like me@greenpeace.org.example.com, which will probably never happen, but eh 🤷 ), you can check from the end of the string with [`substr`](https://www.php.net/substr):
```php
$domain = '@greenpeace.org';
if ( substr( $user->data->user_email, -strlen($domain) ) !== $domain ) {
    // error
}
```
maybe even use [`mb_substr`](https://www.php.net/manual/en/function.mb-substr.php) (same syntax as substr) ? I'm not sure of what is allowed in email addresses, it might be safer.